pkg/cmd/roachtest/cluster.go

```diff
@@ -3010,6 +3011,27 @@ func archForTest(ctx context.Context, l *logger.Logger, testSpec registry.TestSp
 	return arch
 }
 
+// bucketCachedClusterVMs buckets cachedCluster.VMs by provider.
+func bucketCachedClusterVMs(cachedCluster *cloud.Cluster) map[string][]vm.VM {
```

Comment:
nit: This seems like it could be useful beyond cached clusters. Maybe `bucketVMsByProvider` instead?

Reply:
Done